src/load_data.jl

```diff
@@ -53,6 +53,80 @@ function load_data(filename::AbstractString)
+    #          quantity_1 (the measured values)
+    #          quantity_2 (the measured values)
+    #       descriptor (contains references to the axes datasets below)
+    #       axis_1 (contains the values of the axis — see note #1)
+    #       axis_2 (contains the values of the axis - see note #1)
+    #    group2/
+    #       ... same structure as above ...
+    #
+    # Note #1: Any axes referenced by the descriptor can themselves contain
+    #          references to multiple subaxes. This indicates that the axis
+    #          is "unstructured", i.e. that multiple coordinates where changed
+    #          simultaneously.
+
+    datasets    = Dict{String, Dict{String, Vector{Any}}}()
+    descriptors = Dict{String, Vector{Dict{String,Any}}}()
+
+    h5open(filename, "r") do f
+        
+        # Find all of the group names, which will correspond to qubits when
+        # using exp_factory inside of auspex. The default group name is simply
+        # "main".
+        group_names = names(f)
+
+        for group_name in group_names
+            g = f[group_name]
+
+            # Read in the descriptor
+            desc_refs = read(g["descriptor"])
+
+            desc = Vector{Dict{String,Any}}()
+            for desc_ref in desc_refs
+                axis = Dict{String,Any}()
+                if reinterpret(Bool, read(attrs(g[desc_ref])["unstructured"]))
+                    axis_refs      = read(g[desc_ref])
+                    axis["points"] = [read(g[axis_ref]) for axis_ref in axis_refs]
+                    axis["name"]   = [read(attrs(g[axis_ref])["name"]) for axis_ref in axis_refs]
+                    axis["unit"]   = [read(attrs(g[axis_ref])["unit"]) for axis_ref in axis_refs]
+                else
+                    axis["points"] = read(g[desc_ref])
+                    axis["name"]   = read(attrs(g[desc_ref])["name"])
+                    axis["unit"]   = read(attrs(g[desc_ref])["unit"])
+                end
+                # See if we need to get metadata:
+                if "metadata" in names(attrs(g[desc_ref]))
+                    meta_ref = read(attrs(g[desc_ref])["metadata"])
+                    meta_enum_ref = read(attrs(g[desc_ref])["metadata_enum"])
+                    axis["metadata"] = [read(g[meta_enum_ref])[m+1] for m in read(g[meta_ref])]
+                end
+                
+                push!(desc, axis)
+            end
+
+            descriptors[group_name] = desc
+
+            # Read in the actual data
+            datacol_names = names(g["data"])
+            datasets[group_name] = Dict{String, Vector{Any}}()
+            for datacol_name in datacol_names
+              dataset = read(g["data"][datacol_name])
+              # heuristic to read complex numbers stored as HDF5Compounds
+              datasets[group_name][datacol_name] = typeof(dataset[1]) == HDF5.HDF5Compound{2}? [Complex(d.data[1],d.data[2]) for d in dataset] : dataset
```

Comment:
I think `complex` with a lowercase "c" is the usual way to construct a `Complex` value.

This line is somewhat hard to read. It might clean things up to use dispatch to select the appropriate "conversion". i.e. something like:
```julia
convert_dataset(ds) = ds
function convert_dataset(ds::Vector{HDF5.HDF5Compound{2})
    return [complex(d.data[1], d.data[2]) for d in ds]
end
```